lib/ansible/modules/windows/win_blockinfile.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: win_blockinfile
+short_description: Ensure a particular line is in a file, or replace an existing line using a back-referenced regular expression
+version_added: '2.8'
+description:
+- This module will insert/update/remove a block of multi-line text surrounded by customizable marker lines.
+options:
+  path:
+    description:
+      - The path of the file to modify.
+      - Note that the Windows path delimiter C(\) must be escaped as C(\\) when the line is double quoted.
+      - Before Ansible 2.3 this option was only usable as I(dest), I(destfile) and I(name).
+    type: path
+    required: yes
+    aliases: [ dest, destfile, name ]
+  state:
+    description:
+      - Whether the line should be there or not.
+    type: str
+    choices: [ absent, present ]
+    default: present
+  marker:
+    description:
+    - The marker line template.
+    - C({mark}) will be replaced with the values C(in marker_begin) (default="BEGIN") and C(marker_end) (default="END").
+    - Using a custom marker without the C({mark}) variable may result in the block being repeatedly inserted on subsequent playbook runs.
+    type: str
+    default: '# {mark} ANSIBLE MANAGED BLOCK'
+  block:
+    description:
+    - The text to insert inside the marker lines.
+    - If it is missing or an empty string, the block will be removed as if C(state) were specified to C(absent).
+    type: str
+    default: ''
+    aliases: [ content ]
+  insertafter:
+    description:
+      - Used with C(state=present). If specified, the line will be inserted after the last match of specified regular expression. A special value is
+        available; C(EOF) for inserting the line at the end of the file.
```

Comment:
This seems to leave open the possibility of collision...  Probably unlikely, but possible (`EOF` is common with embedded heredoc).  Though i guess the user could prevent by providing anchors to the `EOF` if they wanted to match text (`^\s+EOF` or something).

Reply:
Don't disagree, but I think maintaining similarity to `blockinfile` takes priority.